LEAF_Request_Portal/sources/System.php

```diff
@@ -1096,4 +1098,124 @@ private function updateNexusWithPortalGroups(array $portal_groups, \Orgchart\Gro
         }
 
     }
+
+    /**
+     * Create flag (days) for destruction records
+     *
+     * @param int $flag - number of days to mark a record for destruction
+     *
+     * @return bool
+     */
+    public function setDestructionFlag($flag = null): bool
+    {
+        if ($flag) {
+            $vars = array(':data' => $flag);
+            $strSQL = 'INSERT INTO `settings` (`setting`, `data`)
+            VALUES ("destructionFlag", :data)
+            ON DUPLICATE KEY UPDATE `data`=:data';
+
+            $this->db->prepared_query($strSQL, $vars);
+
+            return true;
+        }
```

Comment:
I assume most if not all of these methods will be used via an api call. Don't we want to start making sure our responses are in a proper json format? I have been using 
                    $return_value = array (
                        'status' => array (
                            'code' => 2,
                            'message' => ''
                        ),
                      'data' => `array of data to be returned`
                    );

Not saying what I have been using is what we ought to use but that is something we should talk about as a team.

Reply:
Yes, please.